File: StarcevMatvey/Term 2/Block 2/Task_1/BlackJack/Deck.cs
namespace BlackJack
{
    public class Deck
    {
        public List<Card> Cards { get; }

        public Deck()
        {
            Cards = new List<Card>();
            string[] suits = { "H", "D", "P", "B" };
            string[] values = { "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K", "A" };
            for (int i = 0; i < 52; i++)
            {
                Cards.Add(new Card(suits[i % 4], values[i % 13]));
            }
        }
    }
}

File: Vadim-Yakshigulov/Semester 2/Blackjack/blackjack-base-lib/src/main/kotlin/blackjack/deck/Deck.kt
package blackjack.deck

import blackjack.card.Card
import blackjack.card.CardFace
import blackjack.card.CardSuit
import blackjack.card.ICard

class Deck : IDeck {
    override val cards: List<ICard>
        get() = _cards
    private val _cards: MutableList<ICard> = mutableListOf()
    init {
        renew()
    }

    override fun renew() {
        _cards.clear()
        for (suit in CardSuit.values())
            for (face in CardFace.values()) {
                _cards.add(Card(suit, face))
            }
    }
}